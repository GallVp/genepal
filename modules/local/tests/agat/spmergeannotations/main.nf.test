nextflow_process {

    name "Test Process AGAT_SPMERGEANNOTATIONS"
    script "../../../../nf-core/agat/spmergeannotations/main.nf"
    process "AGAT_SPMERGEANNOTATIONS"

    tag "modules"
    tag "modules_nfcore"
    tag "agat"
    tag "agat/spmergeannotations"

    // Test to show that agat merge fuses two genes into one
    //
    // Issue #153: <https://github.com/Plant-Food-Research-Open/genepal/issues/153>
    test("Merge two genes into one") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    [
                        file("$baseDir" + '/modules/local/tests/agat/spmergeannotations/testdata/file1.gff', checkIfExists: true), // 10 genes
                        file("$baseDir" + '/modules/local/tests/agat/spmergeannotations/testdata/file2.gff', checkIfExists: true) // 5 genes
                    ]
                ]
                input[1] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()

            def featureLines = file(process.out.gff[0][1]).text.readLines().findAll { ! it.startsWith('#') }
            def geneLines = featureLines.collect { it.split('\t') }.findAll { it[2] == 'gene' }

            with(geneLines) {
                assert size() != 15 // 10 + 5
                assert size() == 14 // 2 genes from file1 were fused into 1
            }
        }

    }

}
