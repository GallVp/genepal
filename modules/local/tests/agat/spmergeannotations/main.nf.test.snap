{
    "Merge two genes into one": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff:md5,7640b96cc77f9b192fc369b98fce0cbf"
                    ]
                ],
                "1": [
                    "versions.yml:md5,41c8654a8b48f0ce58a056ad3333ffeb"
                ],
                "gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff:md5,7640b96cc77f9b192fc369b98fce0cbf"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,41c8654a8b48f0ce58a056ad3333ffeb"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-05T15:16:33.224735"
    }
}